app/code/Magento/CatalogGraphQl/DataProvider/Product/SearchCriteriaBuilder.php

```diff
@@ -104,6 +104,7 @@ public function build(array $args, bool $includeAggregation): SearchCriteriaInte
             $this->addDefaultSortOrder($searchCriteria, $isSearch);
         }
 
+        $this->addEntityIdSort($searchCriteria, $isSearch);
```

Comment:
could you please verify the behavior for GQL query which will filter by category id/ids, e.g.
```
{
  products(filter: {category_id: {in: ["3", "5"]}}) {
   items {    
      sku
    }
  }
}
 
{
  products(filter: {category_id: {eq:"3"}}) {
   items {    
      sku
    }
  }
}
 ```

In these examples we should add order by "position ASC" only for the second case. 
And only if the sort was provided explicitly in GQL query it should be added as is.

I'm afraid that current implementation in addDefaultSortOrder method don't take this into account 

I think we dont' have specific tests for that and it make sense to add this for GQL coverage
(@yaroslavGoncharuk can you help us with steps?)

Reply:
I did test this and because this method (\Magento\Catalog\Model\ResourceModel\Product\Collection::addAttributeToSort line 1749) converts the attribute position to entity_id. I think it makes sense to always have a position set as a sort.